File: src/types/booking.ts
export interface Booking {
  id: string;
  customerName: string;
  tourName: string;
  agencyId: string;
  touristId: string;
  //createdAt: Timestamp;
  price: number; // From AgencyDashboard.tsx
  rating: number; // From AgencyDashboard.tsx
  available: boolean; // From AgencyDashboard.tsx
  image: string; // From AgencyDashboard.tsx
  createdAt?: Date; // From AgencyDashboard.tsx
  category: string; // From AgencyDashboard.tsx
  difficulty: string; // From AgencyDashboard.tsx
  maxParticipants: number; // From AgencyDashboard.tsx
  reviewsCount: number; // From AgencyDashboard.tsx
  tourId: string; // From BookingDetailsModal.tsx
  bookingDate: Date; // From BookingDetailsModal.tsx
  participants: number;
  status: 'confirmed' | 'pending' | 'completed' | 'cancelled';
  totalPrice: number; // From AgencyDashboard.tsx
  customerEmail?: string; // From BookingDetailsModal.tsx (optional)
  paymentStatus: 'Pending' | 'Completed'; // From BookingDetailsModal.tsx
  feedback?: string; // From BookingDetailsModal.tsx (optional)
  specialRequests?: string; // From BookingDetailsModal.tsx (optional)
  tourDate: Date; // From BookingDetailsModal.tsx and AgencyDashboard.tsx (as part of Booking interface in AgencyDashboard for bookings array)
  assignedGuide?: string; // Added based on the update modal
  internalNotes?: string; // Added based on the update modal
}

File: android/app/src/main/java/com/wolaitatours/android/data/model/Booking.kt
package com.wolaitatours.android.data.model

import com.google.firebase.Timestamp
import com.google.firebase.firestore.DocumentSnapshot

enum class BookingStatus {
    pending,
    confirmed,
    completed,
    cancelled,
    disputed,
    review_pending,
    reviewed
}

enum class PaymentStatus {
    pending,
    paid,
    verified,
    failed,
    refunded
}

data class Booking(
    val id: String = "",
    val tourId: String = "",
    val touristId: String = "",
    val agencyId: String = "",
    val agencyName: String? = null,
    val tourName: String = "",
    val customerName: String = "",
    val participants: Int = 1,
    val totalPrice: Double = 0.0,
    val bookingDate: Timestamp? = null,
    val tourDate: Timestamp? = null,
    val status: BookingStatus = BookingStatus.pending,
    val paymentStatus: PaymentStatus = PaymentStatus.pending,
    val paymentId: String? = null,
    val specialRequests: String? = null,
    val completedAt: Timestamp? = null,
    val createdAt: Timestamp? = null, // Added to match website
)

fun DocumentSnapshot.toBooking(): Booking {
    return Booking(
        id = id,
        tourId = getString("tourId").orEmpty(),
        touristId = getString("touristId").orEmpty(),
        agencyId = getString("agencyId").orEmpty(),
        agencyName = getString("agencyName"),
        tourName = getString("tourName").orEmpty(),
        customerName = getString("customerName").orEmpty(),
        participants = (getLong("participants") ?: 1L).toInt(),
        totalPrice = getDouble("totalPrice") ?: 0.0,
        bookingDate = getTimestamp("bookingDate"),
        tourDate = getTimestamp("tourDate"),
        status = runCatching { BookingStatus.valueOf(getString("status").orEmpty()) }
            .getOrDefault(BookingStatus.pending),
        paymentStatus = runCatching { PaymentStatus.valueOf(getString("paymentStatus").orEmpty()) }
            .getOrDefault(PaymentStatus.pending),
        paymentId = getString("paymentId"),
        specialRequests = getString("specialRequests"),
        completedAt = getTimestamp("completedAt"),
        createdAt = getTimestamp("createdAt"), // Added to match website
    )
}

